packages/plugin-typescript/src/lib/typescript-plugin.ts

```diff
@@ -0,0 +1,58 @@
+import { createRequire } from 'node:module';
+import type { PluginConfig } from '@code-pushup/models';
+import { stringifyError } from '@code-pushup/utils';
+import { DEFAULT_TS_CONFIG, TYPESCRIPT_PLUGIN_SLUG } from './constants.js';
+import { createRunnerFunction } from './runner/runner.js';
+import type { DiagnosticsOptions } from './runner/ts-runner.js';
+import { typescriptPluginConfigSchema } from './schema.js';
+import type { AuditSlug } from './types.js';
+import { getAudits, getGroups, logSkippedAudits } from './utils.js';
+
+const packageJson = createRequire(import.meta.url)(
+  '../../package.json',
+) as typeof import('../../package.json');
+
+export type FilterOptions = { onlyAudits?: AuditSlug[] };
+export type TypescriptPluginOptions = Partial<DiagnosticsOptions> &
+  FilterOptions;
```

Comment:
As mentioned before, these types are duplicated and can be replaced with the inferred schema types.

Reply:
removed